x/quarantine/client/testutil/tx_test.go

```diff
@@ -0,0 +1,512 @@
+package testutil
+
+import (
+	"fmt"
+
+	tmcli "github.com/cometbft/cometbft/libs/cli"
+
+	"github.com/cosmos/cosmos-sdk/testutil/cli"
+	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/provenance-io/provenance/testutil/queries"
+	"github.com/provenance-io/provenance/x/quarantine"
+	client "github.com/provenance-io/provenance/x/quarantine/client/cli"
+)
+
+func (s *IntegrationTestSuite) TestTxOptInCmd() {
+	addr0 := s.createAndFundAccount(2000)
+
+	tests := []struct {
+		name    string
+		args    []string
+		expErr  []string
+		expCode int
+	}{
+		{
+			name:   "empty addr",
+			args:   []string{""},
+			expErr: []string{"no to_name_or_address provided"},
+		},
+		{
+			name:   "bad addr",
+			args:   []string{"somethingelse"},
+			expErr: []string{"somethingelse.info: key not found"},
+		},
+		{
+			name:    "good addr",
+			args:    []string{addr0},
+			expCode: 0,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			cmd := client.TxOptInCmd()
+			cmdFuncName := "TxOptInCmd"
+			args := append(tc.args, s.commonFlags...)
+			outBW, err := cli.ExecTestCLICmd(s.clientCtx, cmd, args)
+			out := outBW.String()
+			s.T().Logf("Output:\n%s", out)
+			s.assertErrorContents(err, tc.expErr, "%s error", cmdFuncName)
+			for _, expErr := range tc.expErr {
+				s.Assert().Contains(out, expErr, "%s output with error", cmdFuncName)
+			}
+			if len(tc.expErr) == 0 {
+				var txResp sdk.TxResponse
+				testFuncUn := func() {
+					err = s.clientCtx.Codec.UnmarshalJSON([]byte(out), &txResp)
+				}
+				if s.Assert().NotPanics(testFuncUn, "UnmarshalJSON output") {
+					s.Assert().Equal(tc.expCode, int(txResp.Code), "%s response code", cmdFuncName)
+				}
+			}
+		})
+	}
+}
```

Comment:
Even if the error message isn't as expected, I want the other checks to also run. That way, when it fails, we have all the failure information. That allows the fixer to address all possible problems before running the tests again instead of having to repeatedly run the tests just to figure out what's broken now.

Furthermore, `s.Require().NoError` is not a replacement for `s.assertErrorContents`. The latter asserts `NoError` only if `tc.expErr` is empty or nil. If `tc.expErr` has content, it asserts `Error` and also that the error message contains all the entries in `tc.expErr`.